internal/pkg/service/common/dependencies/base.go

```diff
@@ -17,22 +18,44 @@ import (
 type baseScope struct {
 	logger     log.Logger
 	telemetry  telemetry.Telemetry
+	stdout     io.Writer
+	stderr     io.Writer
```

Comment:
The "one place" is about file downloading to the stdout (or piped bash command).
It is safe to keep stdout opened, it will be closed on the exit.

There is `io.NopCloser` wrapper in the standard library for a reader, but not for a writer.

You can add there a simple wrapper:
https://github.com/keboola/keboola-as-code/blob/9f81b1acfbf820f3142e0e6c7a05c540b2358017/internal/pkg/utils/testhelper/env.go#L161-L168